#! /usr/bin/env python
PACKAGE='lane_detection'

from dynamic_reconfigure.msg import SensorLevels
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       Name                Type      Level     Description                   Default   Min     Max
gen.add("val_thres",        int_t,    0,        "Value threshold",            200,      0,      255)
gen.add("sat_thres",        int_t,    0,        "Saturation threshold",       50,      0,      255)
gen.add("erode_size",       int_t,    0,        "Erosion",                    5,        1,      25)
gen.add("blur_kernel",      int_t,    0,        "Gaussian blur kernel size",  3,        0,      25)
gen.add("mask_height",      int_t,    0,        "Mask over entire image",     420,      0,      480)
gen.add("min_seg_height",   int_t,    0,        "Minimum height of segment",  75,       1,      200)
gen.add("reconstruct_pix",  int_t,    0,        "Vertical pixel sampling",    15,       1,      200)
gen.add("dashed_length",    int_t,    0,        "Dashed line height thresh",  200,      1,      600)
gen.add("fit_tolerance",    double_t, 0,        "Max allowed fit error",      15.0,     1.0,    1000.0)


gen.add("hough_rho_res",    int_t,    0,        "Rho resolution (pix)",       2,        1,      500)
gen.add("hough_theta_res",  double_t, 0,        "Theta resolution (rad)",     0.05,     0.01,   1.0)
gen.add("hough_threshold",  int_t,    0,        "Accumulator threshold",      100,      10,     200)
gen.add("hough_horiz_tol",  double_t, 0,        "Horizontal angle tolerance", 0.1,      0.01,   1.5)

exit(gen.generate(PACKAGE, PACKAGE, "LaneDetection"))
